### PR TITLE
Add Integration test resources

### PR DESCRIPTION
**Pull request by NishikaDeSilva**
**Purpose**

Add resources for Integration Tests

### CODE REVIEW

File: jobs/intg-test-resources/latest/ansible-deployment/copy_deploy_script.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/ansible-deployment/copy_deploy_script.sh:1]: Please add the license headers.

File: jobs/intg-test-resources/latest/ansible-deployment/copy_deploy_script.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+INFRA_PROP_FILE=$INPUT_DIR/infrastructure.properties
+GIT_USER=$(grep -m 1 -w "GIT_WUM_USERNAME" $TEST_PROP_FILE | cut -d'=' -f2)
+GIT_PASS=$(grep -m 1 -w "GIT_WUM_PASSWORD" ${TEST_PROP_FILE} | cut -d'=' -f2)
+keyFileLocation=$(grep -w "keyFileLocation" ${TEST_PROP_FILE} | cut -d'=' -f2)
+
+OS=$(grep  -w "OS" ${INFRA_PROP_FILE} | cut -d'=' -f2)
+WSO2InstanceName=$(grep -w "WSO2InstanceName" ${INFRA_PROP_FILE} | cut -d'=' -f2 | cut -d"/" -f3)
+
+ANSIBLE_REPO=$(grep -w "ANSIBLE_GIT_URL" $TEST_PROP_FILE | cut -d'=' -f2 | cut -d'/' -f3-)
+ANSIBLE_BRANCH=$(grep -w "ANSIBLE_GIT_BRANCH" $TEST_PROP_FILE | cut -d'=' -f2)
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/ansible-deployment/copy_deploy_script.sh:17]: Why cannot we derive this from the product version?

File: jobs/intg-test-resources/latest/ansible-deployment/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/ansible-deployment/deploy.sh:2]: Missing License header.

File: jobs/intg-test-resources/latest/ansible-deployment/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+set -o xtrace
+
+WSO2_PRODUCT_NAME=$1
+WSO2_PRODUCT_VERSION=$2
+GITUSER=$3
+GITPASS=$4
+
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/ansible-deployment/deploy.sh:9]: Please remove unwanted extra lines.

File: jobs/intg-test-resources/latest/ansible-deployment/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+log_info "Updating Database provision scripts"
+sed -i "s/&CF_DB_USERNAME/${DBUsername}/g" ${DB_PROV_SCRIPT_NAME}
+sed -i "s/&CF_DB_PASSWORD/${DBPassword}/g" ${DB_PROV_SCRIPT_NAME}
+sed -i "s/&CF_DB_HOST/${DBHost}/g" ${DB_PROV_SCRIPT_NAME}
+sed -i "s/&CF_DB_PORT/${DBPort}/g" ${DB_PROV_SCRIPT_NAME}
+sed -i "s/&CF_DB_NAME/${DBEngine}/g" ${DB_PROV_SCRIPT_NAME}
+sed -i "s/&CF_DB_VERSION/${DBEngineVersion}/g" ${DB_PROV_SCRIPT_NAME}
+sed -i "s/&PRODUCT_VERSION/${WSO2_PRODUCT_VERSION}/g" ${DB_PROV_SCRIPT_NAME}
+
+bash ${DB_PROV_SCRIPT_NAME}
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/ansible-deployment/deploy.sh:96]: Please add a newline at the EOF.

File: jobs/intg-test-resources/latest/ansible-deployment/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+log_info "Updating ansible resources"
+sed -i "s/DB_TYPE/${DBEngine}/g" ${WORKING_DIR}/${ANSIBLE_REPO_NAME}/roles/common/tasks/main.yml
+sed -i "s/JDK_TYPE/${JDK}/g" ${WORKING_DIR}/${ANSIBLE_REPO_NAME}/roles/common/tasks/main.yml
+sed -i "s/RDS_PWD/${DBPassword}/g" ${WORKING_DIR}/${ANSIBLE_REPO_NAME}/dev/group_vars/*
+sed -i "s/RDS_USERNAME/${DBUsername}/g" ${WORKING_DIR}/${ANSIBLE_REPO_NAME}/dev/group_vars/*
+sed -i "s/DB_ADDRESS/${DBHost}/g" ${WORKING_DIR}/${ANSIBLE_REPO_NAME}/dev/group_vars/*
+sed -i "s/SID/${SID}/g" ${WORKING_DIR}/${ANSIBLE_REPO_NAME}/dev/group_vars/*
+
+log_info "Running ansible scripts"
+cd ${ANSIBLE_REPO_NAME}
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/ansible-deployment/deploy.sh:77]: Is this needed?

File: jobs/intg-test-resources/latest/copy_script.sh [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/copy_script.sh:2]: Please add the License header.

File: jobs/intg-test-resources/latest/test.sh [hunk 8]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/test.sh:2]: License header.

File: jobs/intg-test-resources/latest/test.sh [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+log_info "Copying product pack to Repository"
+cd $LOCAL_PRODUCT_PACK_LOCATION && zip -qr $PRODUCT_NAME-$PRODUCT_VERSION.zip $PRODUCT_NAME-$PRODUCT_VERSION
+mv $PRODUCT_NAME-$PRODUCT_VERSION.zip $PRODUCT_REPOSITORY_PACK_DIR/.
+mv $WORKING_DIR/$NEXUS_SCRIPT_NAME $INT_TEST_MODULE_DIR/.
+
+source /etc/environment
+
+cd $API_IMPORT_EXPORT_MODULE_DIR && mvn clean install -fae -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+cd $INT_TEST_MODULE_DIR  && mvn clean install -s $NEXUS_SCRIPT_NAME -fae -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/test.sh:42]: Missing newline at the EOF.

File: jobs/intg-test-resources/latest/wso2-intg-test-cfn.yaml [hunk 10]
```diff
@@ -3,8 +3,9 @@ Description: AWS CloudFormation Template with an EC2 instance and a RDS
 Parameters:
   AMI:
     Type: String
+    Default: ami-04c21ca1facb0d911
     Description: AMI of the region.
-  DBEngine:
+  DBEngine:
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/wso2-intg-test-cfn.yaml:8]: Please remove extra space.

File: jobs/intg-test-resources/latest/wso2-intg-test-cfn.yaml [hunk 11]
```diff
@@ -235,13 +246,16 @@ Resources:
...
-            "DeviceName" : "/dev/sda1",
-            "Ebs" : { "VolumeSize" : "50" }
-         },
-      ]
+      BlockDeviceMappings:
+        - DeviceName: /dev/sda1
+          Ebs:
+            VolumeSize: '50'
+            VolumeType: gp2
+            DeleteOnTermination: 'true'
```
Comment (VimukthiPerera) [inline jobs/intg-test-resources/latest/wso2-intg-test-cfn.yaml:258]: Shall we write the user-data log to a file similar to scenario tests?

File: jobs/intg-test-resources/latest/test.sh (hunk 9 above)
Comment (msmshariq) [inline jobs/intg-test-resources/latest/test.sh:42]: Add a new line at the end of the file.

File: jobs/intg-test-resources/latest/test.sh [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+mkdir -p $PRODUCT_REPOSITORY_PACK_DIR
+
+log_info "Copying product pack to Repository"
+cd $LOCAL_PRODUCT_PACK_LOCATION && zip -qr $PRODUCT_NAME-$PRODUCT_VERSION.zip $PRODUCT_NAME-$PRODUCT_VERSION
+mv $PRODUCT_NAME-$PRODUCT_VERSION.zip $PRODUCT_REPOSITORY_PACK_DIR/.
+mv $WORKING_DIR/$NEXUS_SCRIPT_NAME $INT_TEST_MODULE_DIR/.
+
+source /etc/environment
+
+cd $API_IMPORT_EXPORT_MODULE_DIR && mvn clean install -fae -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
```
Comment (msmshariq) [inline jobs/intg-test-resources/latest/test.sh:41]: Building `api-import-export` module is specific to API-M, it's not common to all products, need a way to externalize this.

File: jobs/intg-test-resources/latest/ansible-deployment/copy_deploy_script.sh (hunk 2 above)
Comment (NishikaDeSilva) [reply jobs/intg-test-resources/latest/ansible-deployment/copy_deploy_script.sh:17]: Yes, we can. But I thought it's better to add as an input as in [1]. The branch name can be changed anytime. If so this is an easy way of updating scripts.   

[1] https://github.com/NishikaDeSilva/testgrid-job-configs/blob/test-nishi/Integration-Test/wso2am-2.1.0-intg-testgrid.yaml#L42

File: jobs/intg-test-resources/latest/test.sh (hunk 12 above)
Comment (NishikaDeSilva) [reply jobs/intg-test-resources/latest/test.sh:41]: will do

File: jobs/intg-test-resources/latest/ansible-deployment/deploy.sh (hunk 6 above)
Comment (NishikaDeSilva) [reply jobs/intg-test-resources/latest/ansible-deployment/deploy.sh:77]: path in [1] is not absolute.

[1] https://github.com/NishikaDeSilva/ansible-apim-testgrid/blob/2.1.0/dev/group_vars/apim.yml#L22

**Pull request merged**